domain/src/DomainStorage.php

```diff
@@ -94,6 +95,16 @@ public function loadDefaultDomain() {
 
   /**
    * {@inheritdoc}
+   * @return \Drupal\domain\DomainInterface[]
+   */
+  public function loadMultiple(array $ids = NULL) {
```

Comment:
I really think these do not belong here. 
`NodeStorage` does not do this, nor other things in core.
As @agentrickard said, you are merely deferring to the parent and that is an antipattern as far as I kow